### ISSUE TITLE
make Lexycon.py check also lenght, not only format

### ISSUE THREAD

**Issue by belforte** [dmwm/WMCore#10490]
**Impact of the new feature**
Every component which prepares data ultimately targeted for publication in DBS

**Is your feature request related to a problem? Please describe.**
Avoid to get obscure ORA-12899 error late in the workflow

**Describe the solution you'd like**
WMCore/Lexicon.py should validate length of strings used e.g. as LFN's, but in general any string which eventually
will be stored in DBS, instead of checking format only

**Describe alternatives you've considered**
so far I hit this in CRAB dealing with LFN's which contain a user-generated string (those creative users !)
and I have put a length check in my code after validating with Lexicon. It is not very elegant and there
can be places elsewhere in CMS where this check is missing but should be useful.

**Additional context**
yeah.. it is not not urgent, but if there's time and will, it would strengthen Lexicon and possibly avoid errors elsewhere

**Comment by belforte** [comment]
low priority since I have implemented a workaround

**Comment by vkuznet** [comment]
This is related to https://github.com/dmwm/WMCore/issues/10614 and I suggest that we consolidate effort to provide common Lexicon data format that other tools/languages/subsystems can load and use.

**Comment by belforte** [comment]
surely better to do this in the new format. I propose to close

**Comment by amaltaro** [comment]
Or, once we have the new Lexicon model in place, we go there and make this change.
@belforte Stefano, do you know which function is used to validate that? I assume it's `lfn`? What should be the maximum allowed string length?

**Comment by belforte** [comment]
yes Alaan, that's what I meant. As far a limit goes, DBS Oracle table has a limit to 500char on file name. So .. `lfn` and `userLfn` for sure, or any other which is used when new file names are 'crafted'.

**Comment by amaltaro** [comment]
@vkuznet Valentin, given that you know what the database limits are for lfns, can you please edit the `lfn` lexicon validation code to take into consideration the length of the string? If we have any other Lexicon method validating LFNs, we might want to adjust those as well.

I would say, before validating any part of the LFN, we should have a final length validation. If you agree with this implementation now, please take ownership of this issue and move it to work in progress. Thanks

**Comment by vkuznet** [comment]
@amaltaro please provide concrete road map how to deal with lexicon. So far we have:
- https://github.com/dmwm/WMCore/pull/10615
- https://github.com/dmwm/WMCore/pull/10622
- https://github.com/dmwm/WMCore/pull/10764
- and I have lexicon reader/writer json files for DBS Go server
   - DBS reader lexicon https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/preprod/dbs2go-global-r/lexicon.json
   - DBS writer lexicon https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/preprod/dbs2go-global-w/lexicon.json

The lexicon JSON files already have length limits, e.g. see [here](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/preprod/dbs2go-global-w/lexicon.json#L17), and DBS Go server uses them. But it is truly unclear how you'd like to proceed with Python code base. I see few options:
- patch Lexicon.py to introduce LFN length
- apply any specific PR to restructure Lexicon.py according to drafts I proposed and then use the lenggth from JSON files

Once I'll have clear idea how you want to deal with it I can proceed.

**Comment by vkuznet** [comment]
in addition, we can fix https://github.com/dmwm/WMCore/issues/10810, but this again should be done once we decide which approach to choose for Lexicon.py codebase.
